providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -103,7 +103,7 @@ static int params_to_key(DH *dh, const OSSL_PARAM params[])
     return 1;
 
  err:
-    BN_free(priv_key);
+    BN_clear_free(priv_key);
```

Comment:
This is technically unrelated to the rest of the PR, should the commit with this change for dh and dsa go in a separate PR?

Reply:
Is this in a separate commit? If it is I'm ok with it here.

Reply:
It is separated!